src/external/ACSAdapter/createFileScanIngressMiddleware.ts

```diff
@@ -0,0 +1,53 @@
+const createFileScanIngressMiddleware = (): CallableFunction => {
+    const fileScanIngressMiddleware = () => (next: any) => async (activity: any) => {  // eslint-disable-line @typescript-eslint/no-explicit-any
+        const {attachments, channelData} = activity;
+        if (!attachments) {
+            return next(activity);
+        }
+
+        // Patch references
+        if (!channelData?.metadata?.amsreferences && channelData?.metadata?.amsReferences) {
+            activity.channelData.metadata.amsreferences = channelData.metadata.amsReferences;
+        }
+
+        const {metadata: {amsreferences}} = channelData;
+
+        if (amsreferences) {
+            try {
+                const scanFile = (fileId: string) => {
+                    const { fileScanner } = (window as any).chatAdapter.fileManager;  // eslint-disable-line @typescript-eslint/no-explicit-any
+
+                    if (!fileScanner) {
+                        return next(activity);
+                    }
+
+                    const scanResult = fileScanner.retrieveFileScanResult(fileId);
+                    if (scanResult) {
+                        const {scan, fileMetadata} = scanResult;
+                        const index = activity.attachments.findIndex((attachment: any) => (attachment.name === fileMetadata.name)); // eslint-disable-line @typescript-eslint/no-explicit-any
+
+                        if (!activity.channelData.fileScan) {
+                            activity.channelData.fileScan = [];
+                        }
+
+                        activity.channelData.fileScan[index] = scan;

```

Comment:
Isn't this prone to possible out-of-bond errors?

if fileScan is an empty array and index is greater than zero

Reply:
It would not because `JS` handles would handle this particular scenario. If you create an array `[]` (length 0) then add an element to index `2`, then array would simply be turned into `[undefined, undefined, 2]` (length 3) by itself.